content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,675 @@
+
+The above scenario can be replicated by using a [Sidecar Container](/docs/concepts/workloads/pods/sidecar-containers/) as a helper container to write the HTML file.  
+As a Sidecar Container is conceptually an Init Container, it is guaranteed to start before the main webserver container.  
+This ensures that the HTML file is always available when the webserver is ready to serve it.  
+Please see [Enabling sidecar containers](/docs/concepts/workloads/pods/sidecar-containers/#enabling-sidecar-containers) to utilize this feature.
+
+If you are continuing from the previous scenario, you can reuse the ConfigMap named `color` for this scenario.  
+If you are executing this scenario independently, use the `kubectl create configmap` command to create a ConfigMap from [literal values](/docs/tasks/configure-pod-container/configure-pod-configmap/#create-configmaps-from-literal-values):
+
+```shell
+kubectl create configmap color --from-literal=color=blue
+```
+
+Below is an example manifest for a Deployment that manages a set of Pods, each with a main container and a Sidecar Container. The two containers share an `emptyDir` volume that they use to communicate.
+The main container runs a webserver (NGINX). The mount path for the shared volume in the webserver container is `/usr/share/nginx/html`. The second container is a Sidecar Container based on Alpine Linux which acts as a helper container. For this container the `emptyDir` volume is mounted at `/pod-data`. The Sidecar Container writes a file in HTML that has its content based on a ConfigMap. The webserver container serves the HTML via HTTP.
+
+{{% code_sample file="deployments/deployment-with-configmap-and-sidecar-container.yaml" %}}
+
+Create the Deployment:
+```shell
+kubectl apply -f https://k8s.io/examples/deployments/deployment-with-configmap-and-sidecar-container.yaml
+```
+
+Check the pods for this Deployment to ensure they are ready (matching by
+{{< glossary_tooltip text="selector" term_id="selector" >}}):
+```shell
+kubectl get pods --selector=app.kubernetes.io/name=configmap-sidecar-container
+```
+You should see an output similar to:
+```
+NAME                                           READY   STATUS    RESTARTS   AGE
+configmap-sidecar-container-5fb59f558b-87rp7   2/2     Running   0          94s
+configmap-sidecar-container-5fb59f558b-ccs7s   2/2     Running   0          94s
+configmap-sidecar-container-5fb59f558b-wnmgk   2/2     Running   0          94s
+````
+
+Expose the Deployment (the `kubectl` tool creates a
+{{<glossary_tooltip text="Service" term_id="service">}}  for you):
+```shell
+kubectl expose deployment configmap-sidecar-container --name=configmap-sidecar-service --port=8081 --target-port=80
+```
+
+Use ```kubectl``` to forward the port:
+```shell
+kubectl port-forward service/configmap-sidecar-service 8081:8081 & # this stays running in the background
+```
+
+Access the service.
+```shell
+curl http://localhost:8081
```

Comment:
Where do you run this? Does the `localhost` name resolve where people would expect?

Reply:
This is run from a local terminal (through which all the kubectl commands are executed). The localhost name resolves perfectly. I've tested it via minikube.